### PR TITLE
[WFLY-9790] / [WFLY-9791] Remove Servlet Specific Integration from Undertow Subsystem

### PR DESCRIPTION
**Pull request by darranl**
:warning: Do not Rebase this PR, the commits in this PR are contained in further topic branches. :warning: 

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:
- [x] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue(s)

https://issues.jboss.org/browse/WFLY-9790
https://issues.jboss.org/browse/WFLY-9791
https://issues.jboss.org/browse/WFLY-10788

This pull request depends on the following PR first being merged to WildFly Core and tagged - https://github.com/wildfly/wildfly-core/pull/3390

- [x] Pull Request contains description of the issue(s)

This pull request is just a refactoring to clean up the servlet specific integration from the Undertow subsystem into Elytron Web.

- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted

We have discussed this change with QE and it is preferred we get the refactoring merged as a unit ow work and then RFE development can subsequently build upon it.

For bigger changes, major and minor component upgrades make sure your PR also meets following requirements:
- [ ] Pull Request requires a change to the documentation
- [ ] Documentation have been updated accordingly
- [ ] Tests were added to cover changes

For new features ensure as well:
- [ ] Analysis was done
- [ ] Test Plan has been done
- [ ] Tests were verified in advance

If you are not an active contributor of the WildFly project you can request sponsorship by one of the members to help guide you through the process.

### CODE REVIEW

**Comment by jamezp** [comment]
@darranl Looks the version is missing for `undertow-server-servlet`.

File: servlet-feature-pack/pom.xml [hunk 1]
```diff
@@ -176,6 +176,26 @@
...
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.wildfly.security.elytron-web</groupId>
+            <artifactId>undertow-server-servlet</artifactId>
+            <exclusions>
+                <exclusion>
+                    <artifactId>*</artifactId>
+                    <groupId>*</groupId>
```
Comment (jamezp) [inline servlet-feature-pack/pom.xml]: Is there a reason this is listed twice?

Comment (jamezp) [inline servlet-feature-pack/pom.xml]: Is there a reason this is listed twice?

**Comment by darranl** [comment]
Some experimental test runs: -

https://ci.wildfly.org/viewLog.html?buildId=112675&tab=buildResultsDiv&buildTypeId=WF_WildFlyCoreIntegrationExperiments

https://ci.wildfly.org/viewLog.html?buildId=112677&tab=buildResultsDiv&buildTypeId=WF_WildFlyCoreIntegrationExperimentsWindows

**Comment by darranl** [comment]
Another Windows experiment as the last one ran out of disk space: -

https://ci.wildfly.org/viewLog.html?buildId=112694&tab=buildResultsDiv&buildTypeId=WF_WildFlyCoreIntegrationExperimentsWindows

**Comment by darranl** [comment]
Looks like more file system errors on the Windows build - will ignore for now until we have the required Wildfly Core tag.

**Comment by darranl** [comment]
Retest this please

**Comment by kabir** [comment]
Retest this please

**Comment by darranl** [comment]
I have added one more commit to this PR, although not strictly required by this PR a component upgrade to Core will need it.

I did consider if I could send it over in it's own PR but it depends on a file added by this PR so I would effectively be resubmitting this PR anyway.

**Pull request merged**